apps/content_pages/static/content-pages/admin/genericForeignKeyAddChange.js

```diff
@@ -0,0 +1,121 @@
+function disableNotSelectedOptions($contentTypeId) {
+    if ($contentTypeId.val() != "") {
+        $contentTypeId.find(":not(:selected)").prop('disabled', true);
+    };
+};
+
+function getLinkAndClick($, $link, contentTypeIdValue, objectIdValue, link_type) {
+    $.ajax({
+        url: "/api/v1/content-pages/get-content-type-link/",
+        data: {
+            link_type: link_type,
+            model_id: contentTypeIdValue,
+            object_id: objectIdValue,
+        },
+        success: function (response) {
+            $link.attr("href", response.url);
+            $link.off("click");
+            $link.click();
+        },
+        error: function (response) {
+            console.log(response);
+            alert("Не удалось получить ссылку 😞");
+            $link.removeAttr("href");
+        }
+    });
+};
+
+function addLink($, $contentTypeId, $objectId) {
+    const contentTypeIdValue = $contentTypeId.val();
+    const objectIdValue = $objectId.val();
+    const objectId = $objectId.attr("id");
+
+    if (contentTypeIdValue === "") {
+        return;
+    };
+
+    if (objectIdValue === "") {
+        var linkType = "add";
+        var linkId = "add_" + objectId;
+        var imageSrc = "/static/admin/img/icon-addlink.svg";
+        var imageAlt = "Добавить";
+        var linkClass = "related-widget-wrapper-link add-related";
+    } else {
+        var linkType = "change";
+        var linkId = "change_" + objectId;
+        var imageSrc = "/static/admin/img/icon-changelink.svg";
+        var imageAlt = "Изменить";
+        var linkClass = "related-widget-wrapper-link change-related";
+    };
+
+    let $image = $("<img>", { src: imageSrc, alt: imageAlt });
+    let $link = $("<a>", { id: linkId, href: "#", class: linkClass });
+
+    $link = $image.wrap($link).parent();
+    $contentTypeId.after($link);
+
+    $link.click(function (event) {
+        event.preventDefault();
+        event.stopPropagation();
+        getLinkAndClick($, $link, contentTypeIdValue, objectIdValue, linkType);
+    });
+};
+
+function getPrefix($this) {
+    let id = $this.attr("id");
+    let prefix = id.substring(0, id.lastIndexOf("-") + 1);
+    return prefix
+};
+
+function getObjectIdFieldName($this) {
+    let prefix = getPrefix($this);
+    return prefix + "object_id";
+};
+
+function getObjectId($, $this) {
+    const objectIdFieldName = getObjectIdFieldName($this);
+    return $(document.getElementById(objectIdFieldName));
```

Comment:
jQuery умеет обращаться к id с помощью символа `#`. Поэтому, лучше и использовать этот метод;
```suggestion
    return $(`#${objectIdFieldName}`)
```
или 
```suggestion
    return $('#' + objectIdFieldName);
```

Reply:
Вообще мне оч не понравилось получать  `return $('#' + objectIdFieldName);` и осознанно заменил на ванильный js.

Если кажется с `"#" + ` понятнее, то давай поменяю. Менять?

Reply:
Если не нравится, то пусть остаётся так. Потом попросим команду фрота поревьювать, посмотрим на их мнение.